src/components/mobile/barcodeScanner.tsx

```diff
@@ -84,6 +126,29 @@ export class BarcodeScanner extends React.PureComponent<
     );
   };
 
+  private renderAssetView(asset: AssetObject | null) {
+    if (asset) {
+      return (
+        <MobileAssetView
+          {...this.props}
+          token={this.state.token}
+          asset={asset}
+          data_override={() => {
+            const { cpu, display_color, storage, memory_gb } = asset;
+            return { cpu, display_color, storage, memory_gb };
```

Comment:
@clairecfu I wasn't sure how `data_override` works in AssetView but I copied it over to MobileAssetView bc I figured it would still be relevant. Is this correct? Rather than having something like `this.props.asset`, I just pass in the detected asset, which is in `this.state.asset`.